templates/decays.hpp.in

```diff
@@ -0,0 +1,276 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+// File generated at @DateAndTime@
+
+/**
+ * @file @ModelName@_decays.hpp
+ *
+ * @brief contains class for calculating particle decays
+ *
+ * This file was generated at @DateAndTime@ with FlexibleSUSY
+ * @FlexibleSUSYVersion@ (git commit: @FlexibleSUSYGitCommit@) and SARAH @SARAHVersion@ .
+ */
+
+#ifndef @ModelName@_DECAYS_H
+#define @ModelName@_DECAYS_H
+
+#include <boost/core/demangle.hpp>
+
+#include "@ModelName@_decay_table.hpp"
+#include "@ModelName@_mass_eigenstates.hpp"
+#include "@ModelName@_mass_eigenstates_decoupling_scheme.hpp"
+#include "@ModelName@_input_parameters.hpp"
+#include "cxx_qft/@ModelName@_qft.hpp"
+#include "@ModelName@_decay_amplitudes.hpp"
+#include "decays_loop_corrections.hpp"
+#include "decays_problems.hpp"
+#include "effective_couplings.hpp"
+#include "lowe.h"
+#include "wrappers.hpp"
+#include "error.hpp"
+
+namespace flexiblesusy {
+
+template <typename Field1, typename Field2>
+double
+final_state_symmetry_factor(typename @ModelName@_cxx_diagrams::field_indices<Field1>::type const& idx1,
```

Comment:
It seems to me that this function is model independent and may be better put into `src/`.

Reply:
Are you sure? It relies on `fields` which are always model dependent.

Reply:
You are right, sorry, I've overlooked that.